### DISCUSSION TITLE
Selecting only specific costing factors along the route

### DISCUSSION THREAD

**Discussion by Kashian** [valhalla/valhalla#4269] [Q&A]
Hi. 
Hope my question is not naive :)
I am trying to figure out if through json parameters we send to routing engine, we have the option to say for example do not take travel distance/time as a cost and assume it is 0.
The reason for asking this question is that I want to only find a route which has the minimum number of Turns (left - right - u_turn) and I do not care about travel time and the distance between two nodes of the graph. 
So may be travel time doubles, but number of turns is reduced.
I need to know how I can tell the costing module to ignore travel time as a cost factor, so total cost does not include travel time or travel distance.
I am basically focusing on truck routing.

**Comment by Kashian** [comment]
@kevinkreiser  I appreciate your comment on this.

**Reply by nilsnolde** [reply]
I think you’ll still need to have a concept of increasing cost for _every_ edge during the expansion, otherwise I’m pretty sure you won’t even get a route, except for rare or trivial cases. You can try maneuver_penalty and see if that’s enough for you. If you need smth more complex, you’ll have to modify the EdgeCost and or TransitionCost of the truck costing.

**Reply by Kashian** [reply]
@nilsnolde Thanks.
As you say, increasing cost for every edge looks necessary.. But is there any option where I can for example say the factor is 0.1 ...so the weight of all edges will become less than tenth and even if we add it up along the route, still they are not comparable to heavy weight maneuvers. In this case, we only should find the routes that have less maneuvers. 
In the documentation, it says, maneuver penalty only applies after the road name in the next edge is changed. This is not looking a good way of handling maneuvers. cause there are streets that they go all straight, but after passing a red light, the street name changes....it does not mean by that change we had a change of maneuver. We are still going straight. I can provide other examples as well. however I wanted to define maneuvers for certain read world driving maneuvers. like merging into highway, keep left or right at Fork, turn left or right, u turn, and else... I have a list of around 30 maneuvers which are well established in Transport Engineering. So identifying the type of maneuver is essential.
I appreciate your comments.
